src/arithmetic.jl

```diff
@@ -572,6 +575,19 @@ for T in (:Taylor1, :TaylorN)
     end
 end
 
+# in-place product: `a` <- `a*b`
+# this method computes the product `a*b` and saves it back into `a`
+# assumes `a` and `b` are of same order
+function mul!(a::TaylorN{T}, b::TaylorN{T}) where {T<:Number}
```

Comment:
An equivalent method `mul!(a::Taylor1{T}, b::Taylor1{T}) where {T<:Number})` could be useful for the extension of `pow!` for `Taylor1{TaylorN{T}}`?

Reply:
It would indeed by useful, I've added the corresponding method, thanks!